cql3/statements/select_statement.cc

```diff
@@ -1154,49 +1156,82 @@ indexed_table_select_statement::do_execute(query_processor& qp,
+                    if (result_primary_keys_paging_state.has_error()) {
+                        co_return failed_result_to_result_message(std::move(result_primary_keys_paging_state));
+                    }
+                    auto&& [primary_keys, paging_state] = result_primary_keys_paging_state.assume_value();
+                    {
+                        auto result_results_cmd = co_await this->do_execute_base_query(qp, std::move(primary_keys), state, *internal_options, now, paging_state);
+                        if (result_results_cmd.has_error()) {
+                            co_return failed_result_to_result_message(std::move(result_results_cmd));
+                        }
+                        auto&& [results, cmd] = result_results_cmd.assume_value();
+                        {
+                            auto result_stop = consume_results(std::move(results), std::move(cmd), std::move(paging_state));
+                            if (result_stop.has_error()) {
+                                co_return failed_result_to_result_message(std::move(result_stop));
+                            }
+                            stop = std::move(result_stop).assume_value();
+                        }
+                    }
                 }
-            }).then(wrap_result_to_error_message([this, &builder] () {
+            } while (!stop);
+            {
                 auto rs = builder.build();
                 update_stats_rows_read(rs->size());
                 _stats.filtered_rows_matched_total += _restrictions_need_filtering ? rs->size() : 0;
                 auto msg = ::make_shared<cql_transport::messages::result_message::rows>(result(std::move(rs)));
-                return make_ready_future<shared_ptr<cql_transport::messages::result_message>>(std::move(msg));
-            }));
-        });
+                co_return shared_ptr<cql_transport::messages::result_message>(std::move(msg));
+            }
+        }
     }
 
     if (whole_partitions || partition_slices) {
         tracing::trace(state.get_trace_state(), "Consulting index {} for a single slice of keys", _index.metadata().name());
         // In this case, can use our normal query machinery, which retrieves
         // entire partitions or the same slice for many partitions.
-        return find_index_partition_ranges(qp, state, options).then(wrap_result_to_error_message([now, &state, &options, &qp, this] (std::tuple<dht::partition_range_vector, lw_shared_ptr<const service::pager::paging_state>> result) {
-            auto&& [partition_ranges, paging_state] = result;
-            return this->execute_base_query(qp, std::move(partition_ranges), state, options, now, std::move(paging_state));
-        }));
+        coordinator_result<std::tuple<dht::partition_range_vector, lw_shared_ptr<const service::pager::paging_state>>> result = co_await find_index_partition_ranges(qp, state, options);
```

Comment:
You used `auto` above, why not here (and below)?

Reply:
I'm just being super conservative in the transformation. The "before" code had the type spelled out, so so does that "after" code.